viroconcom/plot.py

```diff
@@ -0,0 +1,520 @@
+        self.y_outside = y_outside
+        self.return_period = return_period
+
+
+def plot_confidence_interval(x_median, y_median, x_bottom, y_bottom, x_upper,
+                             y_upper, ax, x_label=None, y_label=None,
+                             contour_labels=[None, None, None], plotted_sample=None):
+    """
+    Plots the confidence interval (median, bottom, upper) in a standardized
+    appearance.
+    Parameters
+    ----------
+    x_median : ndarray of floats
+        The 50-percentile contour's coordinates in the x-direction.
+    y_median : ndarray of floats
+        The 50-percentile contour's coordinates in the y-direction.
+    x_bottom : ndarray of floats
+        The bottom percentile contour's coordinates in the x-direction.
+    y_bottom : ndarray of floats
+        The bottom percentile contour's coordinates in the y-direction.
+    x_upper : ndarray of floats
+        The upper percentile contour's coordinates in the x-direction.
+    y_upper : ndarray of floats
+        The upper percentile contour's coordinates in the y-direction.
+    ax : Axes
+        Axes of the figure where the contour should be plotted.
+    x_label : str, optional (defaults to None)
+        Label for the x-axis.
+    y_label : str, optional (defaults to None)
+        Label for the y-axis.
+    contour_labels : list of str, optional (defaults to [None, None, None])
+        Label for the environmental contours that will be used in the legend.
+    plotted_sample : PlottedSample, optional
+        If provided, this sample is plotted together with the contours.
+    """
+    plot_contour(x=x_median,
+                 y=y_median,
+                 ax=ax,
+                 x_label=x_label,
+                 y_label=y_label,
+                 line_style='b-',
+                 contour_label=contour_labels[0],
+                 plotted_sample=plotted_sample)
+    plot_contour(x=x_bottom,
+                 y=y_bottom,
+                 contour_label=contour_labels[1],
+                 line_style='r--',
+                 ax=ax)
+    plot_contour(x=x_upper,
+                 y=y_upper,
+                 contour_label=contour_labels[2],
+                 line_style='r--',
+                 ax=ax)
+
+
+def plot_wave_breaking_limit(ax, bottom_tz=0, upper_tz=20, steps=100):
+    """
+    Plots the wave breaking limit on a given axes.
+
+    Assumes that x = zero-up-crossing period and y = sig. wave height.
+
+    Parameters
+    ----------
+    ax : Axes
+    bottom_tz : float (defaults to 0)
+        Bottom limit for the curve.
+    upper_tz : float (defaults to 20)
+        Upper limit for the curve.
+    steps : int (defaults to 100)
+        Number of points that are plotted on the curve.
+    """
+    tz_lim = np.linspace(bottom_tz, upper_tz, steps)
+    hs_lim = hs_from_limiting_sig_wave_steepness(tz_lim)
+    ax.plot(tz_lim, hs_lim, linestyle='-.', color=[0.5, 0.5, 0.5])
+
+
+def hs_from_limiting_sig_wave_steepness(tz):

```

Comment:
Strictly speaking, this is not a plot function, thus may or may not belong in the plot module.

Reply:
True.. but do we have a better place for it?